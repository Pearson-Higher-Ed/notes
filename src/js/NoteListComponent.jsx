import React from 'react';
import { IntlProvider } from 'react-intl';
import InternationalSupport from './InternationalSupport';
import NoteList from './NoteList';

export const NoteListComponent = function NoteListComponent(paramsObj) { // eslint-disable-line import/prefer-default-export
  const intlObj = new InternationalSupport(paramsObj.locale);
  return (<IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}>
    <NoteList
      notes={paramsObj.notes}
      clickNoteHandler={paramsObj.clickNoteHandler}
      removeNoteHandler={paramsObj.removeNoteHandler}
      drawerCallbacks={paramsObj.drawerCallbacks} />
  </IntlProvider>)
}
