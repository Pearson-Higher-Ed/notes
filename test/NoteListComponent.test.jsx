/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/

/* eslint-disable */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import NoteListComponent from '../src/js/NoteListComponent';
import msgObject from '../translations';

const intlObj = new InternationalSupport(msgObject, 'en');


const clickNoteHandler = () => {
   console.log('clickNoteHandler');
 };

it('renders correctly', () => {
  const tree = renderer
    .create(<MuiThemeProvider><IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}><NoteListComponent /></IntlProvider></MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();

  console.log(tree);
});

const clickBookmarkHandler = () => {
   console.log('clickBookmarkHandler');
 };
 it('renders renderNote function', () => {
 	const component = renderer.create(
   <IntlProvider locale={intlObj.getLocale()} messages={intlObj.getMessages()}>
   <NoteListComponent
     clickNoteHandler={clickNoteHandler}
     removeNoteHandler={()=>{}}
     /></IntlProvider>
 );
 let tree = component.toJSON();
 expect(tree).toMatchSnapshot(); 
});





