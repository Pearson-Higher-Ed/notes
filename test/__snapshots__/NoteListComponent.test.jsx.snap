// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="notes"
  role="main"
>
  <div
    className="notes-body"
  />
  <div
    aria-live="assertive"
    className="reader-only"
    id="notes-assert-container"
    role="alert"
  />
</div>
`;

exports[`renders renderNote function 1`] = `
<div
  id="notes"
  role="main"
>
  <div
    className="notes-body"
  />
  <div
    aria-live="assertive"
    className="reader-only"
    id="notes-assert-container"
    role="alert"
  />
</div>
`;
